build-scripts/hack/update-contour-components.sh

```diff
@@ -41,6 +41,27 @@ cp contour-src/examples/gateway-provisioner/01-roles.yaml ck-gateway-contour/tem
 cp contour-src/examples/gateway-provisioner/02-rolebindings.yaml ck-gateway-contour/templates/
 cp contour-src/examples/gateway-provisioner/03-gateway-provisioner.yaml ck-gateway-contour/templates/
 
+# change gateway provisioner image to use the values from values.yaml
+sed -i "s|image: ghcr.io/projectcontour/contour:${CONTOUR_VERSION}|image: \"{{ .Values.projectcontour.image.registry }}/{{ .Values.projectcontour.image.repository }}:{{ .Values.projectcontour.image.tag }}\"|" ck-gateway-contour/templates/03-gateway-provisioner.yaml
+
+# Add image args to the gateway provisioner 
+sed -i '/^        - --enable-leader-election$/a\ \ \ \ \ \ \ \ - --envoy-image={{ .Values.envoyproxy.image.registry }}/{{ .Values.envoyproxy.image.repository }}:{{ .Values.envoyproxy.image.tag }}\n\ \ \ \ \ \ \ \ - --contour-image={{ .Values.projectcontour.image.registry }}/{{ .Values.projectcontour.image.repository }}:{{ .Values.projectcontour.image.tag }}' ck-gateway-contour/templates/03-gateway-provisioner.yaml
+
+# Add values.yaml
+cat <<EOF >ck-gateway-contour/values.yaml
+projectcontour:
+  image:
+    registry: ghcr.io
+    repository: projectcontour/contour
```

Comment:
this is unnecessary, just make it `repository` (merge registry/repository) and `tag`. it simplifies your configs as well.